api/ssi-bridge/src/services/ssi-service.ts

```diff
@@ -203,31 +213,40 @@ export class SsiService {
 		}
 	}
 
-	async generateIdentity(): Promise<{ account: Identity.Account; doc: Identity.Document; key: Identity.KeyPair }> {
+	async generateIdentity(): Promise<{
+		doc: Identity.Document;
+		key: Identity.KeyPair;
+		encryption: Identity.KeyPair;
+	}> {
 		try {
-			const builder = this.getAccountBuilder();
-			const keyPair = new KeyPair(KeyType.Ed25519);
-			const account = await builder.createIdentity({ privateKey: keyPair.private() });
+			const signKeyPair = new KeyPair(KeyType.Ed25519);
+			const document = new Document(signKeyPair, this.getConfig(false).network.name());
+
+			// Add encryption keys and capabilities to Identity
+			const encryptionKeyPair = new KeyPair(KeyType.X25519);
+			const encryptionMethod = new Identity.VerificationMethod(
+				document.id(),
+				encryptionKeyPair.type(),
+				encryptionKeyPair.public(),
+				'kex-0'
+			);
+			document.insertMethod(encryptionMethod, MethodScope.KeyAgreement());
+
+			document.signSelf(signKeyPair, document.defaultSigningMethod().id());
```

Comment:
cant we use publishSignedDoc here?

Reply:
I think you need to pass prevMessageId as undefined and add a check into the publishSignedDoc and only set:

`if(prevMessageId) { newDoc.setMetadataPreviousMessageId(prevMessageId);}`